pkg/controller/istiocsr/deployments_test.go

```diff
@@ -808,6 +828,16 @@ func TestUpdateArgList(t *testing.T) {
 					t.Errorf("Expected to find argument %q in container args, but it was not found. Args: %v", expectedArg, containerArgs)
 				}
 			}
+
+			// Verify arguments that should NOT be present
+			for _, argName := range tt.notExpectedArgs {
+				for _, arg := range containerArgs {
+					if strings.HasPrefix(arg, fmt.Sprintf("--%s=", argName)) {
+						t.Errorf("Expected NOT to find argument with prefix %q in container args, but found %q. Args: %v", fmt.Sprintf("--%s=", argName), arg, containerArgs)
```

Comment:
nit: can we update to below?
```suggestion
						t.Errorf("Expected NOT to find argument %q in container args. Args: %v", arg, containerArgs)
```

Reply:
Suggestion is incorporated